### DISCUSSION TITLE
OBS zoom and follow for Mac?

### DISCUSSION THREAD

**Discussion by marcelomoyano** [tryptech/obs-zoom-and-follow#26] [Q&A]
hello! is there a way to set this to work on OBS for Mac?. 

I've tried to install it and chose the python path in OBS but I don't see any parameters in the script window (from OBS)...

Let me know, please! thanks.

**Comment by bpatram** [comment]
I know this is a pretty old question, but I also ran into this issue on macOS. My issue was not specific to this plugin but might be what is causing your issue as well (or anyone else finding this question). I am running OBS 27.2.4, and I have no idea if this is an issue with my configuration or with how OBS loads Python.

I was getting the same issue that you were getting; where I added the script but no parameters would appear. Even if I set the Python Install Path in OBS I still got the issue. I looked at OBS's logs (Help > View Current Log) and noticed this:

```
os_dlopen(libpython3.9.dylib->libpython3.9.dylib): dlopen(libpython3.9.dylib, 0x0101): tried: 
'/Applications/OBS.app/Contents/MacOS/../Frameworks/libpython3.9.dylib' (no such file), 
'/Applications/OBS.app/Contents/MacOS/../Frameworks/libpython3.9.dylib' (no such file), 
'/Applications/OBS.app/Contents/MacOS/../Frameworks/libpython3.9.dylib' (no such file), 
'/Applications/OBS.app/Contents/MacOS/../Frameworks/libpython3.9.dylib' (no such file), 
'libpython3.9.dylib' (no such file), 
'/usr/lib/libpython3.9.dylib' (no such file), 
'//libpython3.9.dylib' (no such file)
``` 

So it seems that OBS was not able to find the Python module. It also is looking specifically for Python 3.9 in this version. So to "fix" the issue I added a symlink to the actual libpython3.9.dylib file to the first path OBS looks for it at (inside the OBS app package Frameworks directory). There's a few different places where the actual dylib file could be on your system however. 

-  If you installed Python 3.9 via Python's installer you should be able to add the symlink like this:

	  ```sh
	  ln -s /Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/libpython3.9.dylib  /Applications/OBS.app/Contents/Frameworks/libpython3.9.dylib
	  ```

-  If you installed Python 3.9 via homebrew _and also ran brew link_ you should be able to add the symlink like this:

	  ```sh
	  ln -s /usr/local/Frameworks/Python.framework/Versions/3.9/lib/libpython3.9.dylib  /Applications/OBS.app/Contents/Frameworks/libpython3.9.dylib
	  ```

After adding the symlink, restart OBS and check the log again. Hopefully that error is gone and when you navigate to scripts you'll see the parameters now. Although, if you got this far I am going to assume you will get this error:

```
[zoom_and_follow_mouse.py] ModuleNotFoundError: No module named 'pywinctl'
```

To fix this you should just need to install that dependency via pip.

```sh
python3 -m pip install pywinctl
```

From there restart OBS once more. You'll most likely get a system alert asking to allow OBS to use accessibility features. You'll need to enable this in System Preferences. After enabling OBS restart it once more. Hopefully now you should see no errors at all and can configure the script and OBS as you would expect!

**Reply by marcelomoyano** [reply]
Thanks for your help. Been waiting for this fix without even realizing how much time it passed already :).

**Reply by bpatram** [reply]
I had mixed luck using this plugin on macOS in its current state and had to make some modifications to it to get at least display capture working as intended: https://github.com/tryptech/obs-zoom-and-follow/pull/32. I hope this helps!

**Comment by aronko** [comment]
Hey @bpatram , this helped me as well! I was able to run Python 3.9 with the script running. However, I'm running into a few issues when I configure the script, was wondering if you're seeing the same thing: 

- No way to select the zoom source (Manually input the source name)
- No way to pick monitor (which I think you're adressing in #32 ?)

These are the options I'm seeing: https://share.getcloudapp.com/wbuzx5ZD 

Most importantly, when I press the shortcut, it zooms _way too much_. It seems like you were able to fix this on your side. Did you have to make edits in the code itself so the script can correctly calculate the new source size? 

Appreciate all the help here, would love to get this working on MacOS!

**Reply by bpatram** [reply]
Hi @aronko I think the two issues you listed should be working in my PR #32. I recently updated my branch so it can support more than one monitor. That branch should also fix the issue you have with selecting the "zoom source" setting as well.

> Most importantly, when I press the shortcut, it zooms way too much. It seems like you were able to fix this on your side. Did you have to make edits in the code itself so the script can correctly calculate the new source size?

The "zoom window width" and "zoom window height" settings should let you adjust how 'far' it zooms in. But, don't forget to set up the transformation properties on your source in OBS so it stretches it to fit correctly. This tutorial (albeit slightly dated) can walk you through what you should set it to https://www.youtube.com/watch?v=OUyPpol0yYo&t=282s (at 4:42).

**Comment by Wizastudios** [comment]
Hi

So I just tried to install this using Python 3.10. I'm going to assume the script isn't made for that since it didn't work. I'd like it if the Read Me had MacOS instructions.

**Reply by tryptech** [reply]
The Readme does state, as of 11 days ago in commit https://github.com/tryptech/obs-zoom-and-follow/commit/197c9f8dbe48f2f9c770c2059d5d4ecaedff407d, to use python version 3.6.x for Windows and MacOS for OBS version <28.
